src/debug_utils.cc

```diff
@@ -299,6 +299,106 @@ void CheckedUvLoopClose(uv_loop_t* loop) {
   CHECK(0 && "uv_loop_close() while having open handles");
 }
 
+std::vector<std::string> NativeSymbolDebuggingContext::GetLoadedLibraries() {
+  std::vector<std::string> list(0);
+#ifdef __linux__
+  dl_iterate_phdr(
+      [](struct dl_phdr_info* info, size_t size, void* data) {
+        std::vector<std::string>* list = static_cast<std::vector<std::string>*>(data);
+        if (info->dlpi_name != nullptr && *info->dlpi_name != '\0') {
+          list->push_back(info->dlpi_name);
+        }
+        return 0;
+      },
+      &list);
+#elif __APPLE__
+  int i = 0;
+  for (const char* name = _dyld_get_image_name(i); name != nullptr;
+       name = _dyld_get_image_name(++i)) {
+    list.push_back(name);
+  }
+
+#elif _AIX
+  // We can't tell in advance how large the buffer needs to be.
+  // Retry until we reach too large a size (1Mb).
+  const unsigned int buffer_inc = 4096; 
+  unsigned int buffer_size = buffer_inc;
+  MallocedBuffer<char> buffer(buffer_size);
+  int rc = -1;
+  while (rc != 0 && buffer_size < 1024 * 1024) { 
+    rc = loadquery(L_GETINFO, buffer.data, buffer_size);
+    if (rc == 0) { 
+      break;  
+    }
+    buffer_size += buffer_inc;
+    buffer = MallocedBuffer<char>(buffer_size);
+  }
+  if (rc == 0) { 
+    char* buf = buffer.data;
+    ld_info* cur_info = nullptr;
+    do {    
+      std::ostringstream str;
+      cur_info = reinterpret_cast<ld_info*>(buf);
```

Comment:
Is this safe w.r.t. alignment? I think the answer is 'yes' because higher-end PPC supports unaligned accesses but I ask anyway.

Reply:
yes, contemporary PPC works fine with unaligned memory, though with an affected performance but no `SIGBUS`. However, given the source is a `MallocedBuffer`, forcing it to align to the target structure may incur some effort that balances the alignment cost?